src/jlgen.jl

```diff
@@ -480,8 +480,32 @@ end # HAS_INTEGRATED_CACHE
 ## codegen/inference integration
 
 function ci_cache_populate(interp, cache, mi, min_world, max_world)
-    if VERSION >= v"1.12.0-DEV.15"
-        inferred_ci = CC.typeinf_ext_toplevel(interp, mi, CC.SOURCE_MODE_FORCE_SOURCE) # or SOURCE_MODE_FORCE_SOURCE_UNCACHED?
+    @static if VERSION >= v"1.12.0-DEV.1434"
+        # see typeinfer.jl: typeinf_ext_toplevel
+        ci = CC.typeinf_ext(interp, mi, CC.SOURCE_MODE_NOT_REQUIRED)
+        inspected = IdSet{CodeInstance}()
+        tocompile = CodeInstance[ci]
+        while !isempty(tocompile)
+            callee = pop!(tocompile)
+            callee in inspected && continue
+            push!(inspected, callee)
+            # now make sure everything has source code, if desired
+            mi = CC.get_ci_mi(callee)
+            def = mi.def
+            if CC.use_const_api(callee)
+                src = CC.codeinfo_for_const(interp, mi, ci.rettype_const)
+            elseif isa(def, Method) && ccall(:jl_get_module_infer, Cint, (Any,), def.module) == 0 && !trim
```

Comment:
Where is `trim` from?

Reply:
Right, badly copied, good catch.

Reply:
You also probably don't want to support running uninferred code on the GPU (aka you need trim==true)
```suggestion
            elseif false
```